web/src/features/datasets/server/dataset-router.ts

```diff
@@ -140,13 +144,44 @@ export const datasetRouter = createTRPCRouter({
       }),
     )
     .query(async ({ input, ctx }) => {
+      const scoresByRunId = await ctx.prisma.$queryRaw<
+        Array<{ scores: Array<ScoreSimplified>; runId: string }>
+      >(Prisma.sql`
+        SELECT
+          runs.id "runId",
+          array_agg(s.score) AS "scores"
```

Comment:
does array agg produce an empty array in case of no scores / did you check this state?